packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -571,9 +572,8 @@ const VirtualListBaseFactory = (type) => {
 		// Native only
 		scrollToPosition (x, y, rtl = this.props.rtl) {
 			if (this.containerRef.current) {
-				this.containerRef.current.scrollTo(
-					(rtl && !this.isPrimaryDirectionVertical) ? this.scrollBounds.maxLeft - x : x, y
-				);
+				const rtlCompensator = (platform.ios || platform.safari) ? 0 : this.scrollBounds.maxLeft;
+				this.containerRef.current.scrollTo((rtl && !this.isPrimaryDirectionVertical) ? rtlCompensator - x : x, y);
```

Comment:
I know the condition check for `this.isPrimaryDirectionVertical` is not newly added one, but I'm not sure this condition is mandatory. Could you plase check it?

Reply:
this isPrimaryDirectionVertical was not required. I excluded this.